File: orac/attempt1/2023/raffle/raffle.cpp
#include <cstdio>

int main() {
  FILE *inFile = fopen("rafflein.txt", "r");
  FILE *outFile = fopen("raffleout.txt", "w");

  int numCompetitors;
  int maxNumber;

  fscanf(inFile, "%d %d\n", &numCompetitors, &maxNumber);

  int entryLog[100001] = { 0 };

  for (int i = 0; i < numCompetitors; i++) {
    int currentNum;
    fscanf(inFile, "%d", &currentNum);
    entryLog[currentNum]++;
  }

  for (int i = 1; i <= maxNumber; i++) {
    printf("%d: %d\n", i, entryLog[i]);
    if (entryLog[i] == 1) {
      // printf("WINNER!");
      fprintf(outFile, "%d", i);
      return 0;
    }
  }

  fprintf(outFile, "-1");
  return 0;
}

File: orac/attempt1/2023/raffle/raffle.py
inputFile = open("rafflein.txt", "r")
outputFile = open("raffleout.txt", "w")

firstLine = inputFile.readline()

numCompetitors = int(firstLine.split(" ")[0])
maxNumber = int(firstLine.split(" ")[1])

secondLine = inputFile.readline().split()

raffle = [0] * maxNumber

for num in secondLine:
  raffle[int(num) - 1] += 1

for i in range(len(raffle)):
  if raffle[i] == 1:
    line = str(i + 1)
    outputFile.write(line)
    exit()

outputFile.write("-1")